monkeytype/cli.py

```diff
@@ -121,9 +122,16 @@ def apply_stub_handler(args: argparse.Namespace, stdout: IO, stderr: IO) -> None
     mod = importlib.import_module(module)
     src_path = inspect.getfile(mod)
     src_dir = os.path.dirname(src_path)
-    pyi_name = module.split('.')[-1] + '.pyi'
+
     with tempfile.TemporaryDirectory(prefix='monkeytype') as pyi_dir:
-        pyi_path = os.path.join(pyi_dir, pyi_name)
+        if src_path.endswith('__init__.py'):
+            pyi_name = '__init__.pyi'
+            pyi_dir = os.path.join(pyi_dir, module)
```

Comment:
This is more complex than it needs to be. We don't need to create a subdirectory at all, we just need to use `__init__.pyi` as the filename. Because when we invoke retype, we pass the final file-containing directory anyway, it doesn't know or care what directories are above that.

Reply:
The build is failing because mypy is complaining about `mkdir_p` not being type annotated. But with this simplification we can get rid of `mkdir_p` and make that problem go away :)

Reply:
Sure.  I thought it might be confusing to not maintain the directory structure, but since it's just a temp folder it should be ok to overwrite files with same names, etc.

Reply:
The temp directory doesn't live longer than one invocation of this function (it is cleaned up at the end of the `with` block), and for one invocation of this function we write exactly one file. So there's no possibility of even overwriting anything.

Reply:
That's how i thought the context manager worked, but still saw folders on disk (?)

Reply:
Odd, shouldn't stick around. Maybe if there was a crash inside the context manager?